marshal.go

```diff
@@ -110,6 +111,32 @@ func (d *Decoder) parseInt() int {
 	return result
 }
 
+func (d *Decoder) parseBignum() *big.Int {
+	sign, _ := d.r.ReadByte()
+
+	// "A long indicating the number of bytes of Bignum data follows,
+	// divided by two. Multiply the length by two to determine the
+	// number of bytes of data that follow."
+	// https://docs.ruby-lang.org/en/master/doc/marshal_rdoc.html#label-Bignum
+	bytelen := d.parseInt() * 2
+	bytes := make([]byte, bytelen)
+
+	// Read the bytes in reverse order so they're big-endian as required
+	// by the big.Int type.
+	for i := 0; i < bytelen; i++ {
+		b, _ := d.r.ReadByte()
+		bytes[bytelen-(i+1)] = b
+	}
```

Comment:
What about if we encoded this "work from the high index to the low index" logic into the loop itself?
```suggestion
	for i := bytelen-1; i >= 0; i-- {
		b, _ := d.r.ReadByte()
		bytes[i] = b
	}
```

Reply:
Good idea, thanks!